app/connectors/DownloadDataConnector.scala

```diff
@@ -59,9 +61,10 @@ class DownloadDataConnector @Inject() (config: FrontendAppConfig, httpClient: Ht
         .map { response =>
           response.status match {
             case OK => Some(response.json.as[DownloadDataSummary])
+            case _  => None
           }
         }
-        .recover { case _: NotFoundException =>
+        .recover { case _: UpstreamErrorResponse =>
```

Comment:
we only want to do not found if there is a not found exception - else we want to let global error handle it